src/pczt.rs

```diff
@@ -0,0 +1,406 @@
+    /// separate because they might need to be independently redacted. (For which role?)
+    pub(crate) rho: Option<Rho>,
+
+    /// The seed randomness for the note being spent.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover.
+    pub(crate) rseed: Option<RandomSeed>,
+
+    /// The full viewing key that received the note being spent.
+    ///
+    /// - This is set by the Updater.
+    /// - This is required by the Prover.
+    pub(crate) fvk: Option<FullViewingKey>,
+
+    /// A witness from the note to the bundle's anchor.
+    ///
+    /// - This is set by the Updater.
+    /// - This is required by the Prover.
+    pub(crate) witness: Option<MerklePath>,
+
+    /// The spend authorization randomizer.
+    ///
+    /// - This is chosen by the Constructor.
+    /// - This is required by the Signer for creating `spend_auth_sig`, and may be used to
+    ///   validate `rk`.
+    /// - After`zkproof` / `spend_auth_sig` has been set, this can be redacted.
+    pub(crate) alpha: Option<pallas::Scalar>,
+
+    /// The ZIP 32 derivation path at which the spending key can be found for the note
+    /// being spent.
+    pub(crate) zip32_derivation: Option<Zip32Derivation>,
+
+    /// The spending key for this spent note, if it is a dummy note.
+    ///
+    /// - This is chosen by the Constructor.
+    /// - This is required by the IO Finalizer, and is cleared by it once used.
+    /// - Signers MUST reject PCZTs that contain `dummy_sk` values.
+    pub(crate) dummy_sk: Option<SpendingKey>,
+
+    /// Proprietary fields related to the note being spent.
+    pub(crate) proprietary: BTreeMap<String, Vec<u8>>,
+}
+
+/// Information about an Orchard output within a transaction.
+#[derive(Getters)]
+#[getset(get = "pub")]
+pub struct Output {
+    /// A commitment to the new note being created.
+    pub(crate) cmx: ExtractedNoteCommitment,
+
+    /// The transmitted note ciphertext.
+    ///
+    /// This contains the following PCZT fields:
+    /// - `ephemeral_key`
+    /// - `enc_ciphertext`
+    /// - `out_ciphertext`
+    pub(crate) encrypted_note: TransmittedNoteCiphertext,
+
+    /// The address that will receive the output.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover.
+    pub(crate) recipient: Option<Address>,
+
+    /// The value of the output.
+    ///
+    /// This may be used by Signers to verify that the value matches `cv`, and to confirm
+    /// the values and change involved in the transaction.
+    ///
+    /// This exposes the value to all participants. For Signers who don't need this
+    /// information, we can drop the values and compress the rcvs into the bsk global.
+    pub(crate) value: Option<NoteValue>,
+
+    /// The seed randomness for the output.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover, instead of disclosing `shared_secret` to them.
```

Comment:
- Remove `shared_secret` reference.
- Describe how the Signer can use `rseed` and `recipient` to decrypt and verify `enc_ciphertext` (in the case where `ock` is not present, or `OvkPolicy::None` was used to make the data unrecoverable from chain by the sender).